### ISSUE TITLE
Markdown Table doesn't support CJK characters

### ISSUE THREAD

**Issue by avasconcelos114** [mattermost/mattermost-mobile#1534]
#### Summary
When parsing a markdown table that includes a Chinese, Japanese, or Korean character, the table doesn't render.

#### Environment Information
- Device Name: Samsung Galaxy S7 (Simulator using Genymotion)
- OS Version: Android 7.1.1
- Mattermost App Version: master branch, latest commit being https://github.com/mattermost/mattermost-mobile/commit/ebbbd9689ba84623a46dc300d59e31316cd8a550
- Mattermost Server Version: 4.8.0

#### Steps to reproduce
1. Create a table with any amount of columns and rows
2. Include a Korean, Chinese, or Japanese character in any cell

#### Expected behavior
Should render the markdown table with CJK characters included

#### Observed behavior
Only the header renders, but the original text is seen below
![md_table_cjk](https://user-images.githubusercontent.com/29700158/37758928-53bc6bd6-2df5-11e8-807d-247e416b317e.png)
Note: above issue happened when the Korean characters "버스" were added to the second row of the table, the same happened when testing when adding random Chinese or Japanese characters. It otherwise renders fine when the whole table only uses the alphabet
#### Possible fixes
I'm unfamiliar with how markdown parsing works, but running `console.log` on [this](https://github.com/hmhealey/commonmark.js/blob/719d8f721d301de3a6bca70562c5fb7470994cd1/lib/blocks.js#L724) function revealed that the rows containing CJK characters weren't even logged.

**Comment by hmhealey** [comment]
Hi @avasconcelos114,

Thanks for the report. Are you able to post the original message as it appears when typed in the textbox or when you edit the post? I tried recreating what you typed, and it's working fine for me. I'm on iOS, but that shouldn't matter for this.

~~~
|No|Category|Line/Stop|
|--|--|--|
|1|Stop|School Bus Stop (LGVillage)|
|2|버스|Premiere Bus stop(Mangpo 3rd exit)|
~~~

<img width="375" alt="screen shot 2018-03-22 at 10 29 45 am" src="https://user-images.githubusercontent.com/3277310/37776615-0f5c523c-2dbc-11e8-99cf-1a19ceff82ed.png">
<img width="379" alt="screen shot 2018-03-22 at 10 29 54 am" src="https://user-images.githubusercontent.com/3277310/37776622-136da2fe-2dbc-11e8-85d1-9a56c57b66a6.png">

**Comment by hmhealey** [comment]
My apologies. I tested all these including the one I sent originally on a Nexus 5, and it does appear to work differently on Android than iOS. I really did not expect that.

I've filed a ticket to track this on our end ([link](https://mattermost.atlassian.net/browse/MM-9940))

**Comment by hmhealey** [comment]
Wow, nice work. The behaviour they describe in that ticket is bizarre, but I think that makes sense. I'll take a look at implementing it since I think we might do a 1.7.1 to fix that.

That thread also mention it should be fixed by upgrading React Native which we're currently doing for 1.8, but we probably wouldn't want to do that big of an upgrade for a dot release

**Comment by avasconcelos114** [comment]
Ah great! 

Just tested it again with your newest commit and it's working nicely :)

Thank you for taking care of it

**Comment by hmhealey** [comment]
Awesome, glad to hear it works. If you're using the version from the App Store or Play Store, it'll be fixed shortly in 1.7.1
